core/src/main/java/org/elasticsearch/index/query/morelikethis/Item.java

```diff
@@ -0,0 +1,339 @@
+                .selectedFields(fields)
+                .routing(routing)
+                .version(version)
+                .versionType(versionType)
+                .perFieldAnalyzer(perFieldAnalyzer)
+                .positions(false)  // ensures these following parameters are never set
+                .offsets(false)
+                .payloads(false)
+                .fieldStatistics(false)
+                .termStatistics(false)
+                .dfs(false);
+        // for artificial docs to make sure that the id has changed in the item too
+        if (doc != null) {
+            termVectorsRequest.doc(doc, true);
+            this.id(termVectorsRequest.id());
+        }
+        return termVectorsRequest;
+    }
+
+    /**
+     * Parses and returns the given item.
+     */
+    public static Item parse(XContentParser parser, Item item) throws IOException {
+        XContentParser.Token token;
+        String currentFieldName = null;
+        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+            if (token == XContentParser.Token.FIELD_NAME) {
+                currentFieldName = parser.currentName();
+            } else if (currentFieldName != null) {
+                if (ParseFieldMatcher.STRICT.match(currentFieldName, Field.INDEX)) {
+                    item.index = parser.text();
+                } else if (ParseFieldMatcher.STRICT.match(currentFieldName, Field.TYPE)) {
+                    item.type = parser.text();
+                } else if (ParseFieldMatcher.STRICT.match(currentFieldName, Field.ID)) {
+                    item.id = parser.text();
+                } else if (ParseFieldMatcher.STRICT.match(currentFieldName, Field.DOC)) {
+                    item.doc(jsonBuilder().copyCurrentStructure(parser));
+                } else if (ParseFieldMatcher.STRICT.match(currentFieldName, Field.FIELDS)) {
+                    if (token == XContentParser.Token.START_ARRAY) {
+                        List<String> fields = new ArrayList<>();
+                        while (parser.nextToken() != XContentParser.Token.END_ARRAY) {
+                            fields.add(parser.text());
+                        }
+                        item.fields(fields.toArray(new String[fields.size()]));
+                    } else {
+                        throw new ElasticsearchParseException(
+                                "failed to parse More Like This item. field [fields] must be an array");
+                    }
+                } else if (ParseFieldMatcher.STRICT.match(currentFieldName, Field.PER_FIELD_ANALYZER)) {
+                    item.perFieldAnalyzer(TermVectorsRequest.readPerFieldAnalyzer(parser.map()));
+                } else if ("_routing".equals(currentFieldName) || "routing".equals(currentFieldName)) {
+                    item.routing = parser.text();
+                } else if ("_version".equals(currentFieldName) || "version".equals(currentFieldName)) {
```

Comment:
ParseField handles camelCase and underscore names, but I don't think it does for this specific case, when the name starts with a `_`.